geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentKafkaDataStoreFactory.scala

```diff
@@ -54,14 +62,44 @@ object ConfluentKafkaDataStoreFactory extends GeoMesaDataStoreInfo with LazyLogg
   override val DisplayName = "Confluent Kafka (GeoMesa)"
   override val Description = "Confluent Apache Kafka\u2122 distributed log"
 
-  val SchemaRegistryUrl = new GeoMesaParam[URL]("kafka.schema.registry.url", "URL to a confluent schema registry server, used to read Confluent schemas (experimental)")
+  val SchemaRegistryUrl =
+    new GeoMesaParam[URL](
+      "kafka.schema.registry.url",
+      "URL to a confluent schema registry server, used to read Confluent schemas (experimental)",
+      optional = false
+    )
+
+  val SchemaOverrides =
+    new GeoMesaParam[String](
+      "kafka.schema.overrides",
+      "Typesafe configuration defining a map from topic name to schema override (experimental)",
+      optional = true,
+      largeText = true
+    )
 
   override val ParameterInfo: Array[GeoMesaParam[_ <: AnyRef]] =
-    KafkaDataStoreFactory.ParameterInfo.+:(SchemaRegistryUrl)
+    KafkaDataStoreFactory.ParameterInfo ++ Array(SchemaRegistryUrl, SchemaOverrides)
 
   override def canProcess(params: java.util.Map[String, _ <: java.io.Serializable]): Boolean = {
-    KafkaDataStoreFactoryParams.Brokers.exists(params) &&
-        KafkaDataStoreFactoryParams.Zookeepers.exists(params) &&
-        SchemaRegistryUrl.exists(params)
+    KafkaDataStoreParams.Brokers.exists(params) &&
+      KafkaDataStoreParams.Zookeepers.exists(params) &&
+      SchemaRegistryUrl.exists(params)
+  }
+
+  private[confluent] def parseSchemaOverrides(schemaOverridesConfig: String): Map[String, (SimpleFeatureType, Schema)] = {
+    try {
+      Option(schemaOverridesConfig)
+        .map(ConfigFactory.parseString(_).resolve())
+        .filter(_.hasPath("schemas"))
+        .map {
+          _.getObject("schemas").asScala.map { case (topic, schemaConfig) =>
+            val schema = new Schema.Parser().parse(schemaConfig.render(ConfigRenderOptions.concise()))
+            topic -> (AvroSimpleFeatureTypeUtils.schemaToSft(schema, Some(topic)), schema)
```

Comment:
Ok, so in the case that all the configuration is correct, this is fine. 

What kind of error messages does one get if the config for one or more of the schemas is wrong?  Is there anyway to give feedback about which schema(s) could be better?

Reply:
I'll make it so it throws errors on a per schema basis